### PR TITLE
[#5] Added donut chart, improved bar chart readability

### PR DESCRIPTION
**Pull request by taahamahdi**

### CODE REVIEW

File: public/index.erb [hunk 1]
```diff
@@ -15,67 +25,127 @@ window.onload = function () {
...
-        options: {
-            title: {
-                display: true,
-                text: chartOptions.title,
-                fontColor: COLOR.WHITE
-            },
-            legend: {
-                display: false
+
+    if (chartName === "commonlyUsedWords" ||
```
Comment (Brainicism) [inline public/index.erb]: We can pass the slice amount through chartOptions instead of doing this check here

File: public/index.erb [hunk 2]
```diff
@@ -15,67 +25,127 @@ window.onload = function () {
...
+            chartName === "perThread" ||
+            chartName === "reactionsByUse" ||
+            chartName === "gamesPlayCount")
+        dataValues = dataValues.slice(0, 25);
+    else if (chartOptions.type === 'doughnut')
+        dataValues = dataValues.slice(0, 10);
+
+    // Shorten visible values/labels to improve chart readability
+
+    if (chartName === "commonlyUsedWords" ||
```
Comment (Brainicism) [inline public/index.erb]: Same here

File: public/index.erb [hunk 3]
```diff
@@ -15,67 +25,127 @@ window.onload = function () {
...
+            chartName === "perThread" ||
+            chartName === "reactionsByUse" ||
+            chartName === "gamesPlayCount")
+        dataLabels = dataLabels.slice(0, 25);
+    else if (chartOptions.type === 'doughnut')
+        dataLabels = dataLabels.slice(0, 10);
+
+    const x_axis_autoskip = (chartOptions) => {
```
Comment (Brainicism) [inline public/index.erb]: js naming convention is camelCase

File: public/index.erb [hunk 4]
```diff
@@ -15,67 +25,127 @@ window.onload = function () {
...
+        if (chartOptions.type === 'line') return true;
```
Comment (Brainicism) [inline public/index.erb]: Can simplify to `return chartOptions.type === 'line'`

Comment (taahamahdi) [reply public/index.erb]: Whoops! Thanks :]

File: public/index.erb (hunk 3 above)
Comment (taahamahdi) [reply public/index.erb]: :+1:

**Comment by Brainicism** [comment]
[#5] and [#6]

**Pull request merged**